bundles/catalogue/metadatasearch/view/MetadataSearchContainer.jsx

```diff
@@ -0,0 +1,92 @@
+
+const Description = () => {
+    return <div>{Oskari.getMsg(METADATA_BUNDLE_LOCALIZATION_ID, 'metadataSearchDescription')}</div>;
+};
+
+const SearchContainer = (props) => {
+    const { query, onChange, onSearch } = props;
+    return <div>
+        <SearchInput
+            value={query}
+            onChange={(event) => onChange(event.target.value)}
+            placeholder={Oskari.getMsg(METADATA_BUNDLE_LOCALIZATION_ID, 'placeholder')}
+            onSearch={onSearch}/>
+    </div>;
+};
+
+SearchContainer.propTypes = {
+    query: PropTypes.string,
+    onChange: PropTypes.func,
+    onSearch: PropTypes.func
+};
+
+const MetadataSearchContainer = ({ state, controller }) => {
+    const { query,
+        advancedSearchExpanded,
+        advancedSearchOptions,
+        advancedSearchValues,
+        loading,
+        searchResultsVisible,
+        searchResults,
+        searchResultsFilter,
+        displayedCoverageId,
+        drawing,
+        selectedLayers } = state;
+    return <div>
+        { loading && <FlexRowCentered><Spin/></FlexRowCentered>}
+        {
+            !(loading || searchResultsVisible) &&
+            <>
+                <Description/>
+                <SearchContainer query={query} onChange={controller.updateQuery} onSearch={controller.doSearch}/>
+                <AdvancedSearchContainer
+                    isExpanded={advancedSearchExpanded}
+                    toggleAdvancedSearch={controller.toggleAdvancedSearch}
+                    advancedSearchOptions={advancedSearchOptions}
+                    advancedSearchValues={advancedSearchValues}
+                    controller={controller}
+                    drawing={drawing}/>
+            </>
+        }
+        {
+            (!loading && searchResultsVisible) &&
+            <>
+                <MetadataSearchResultListContainer
+                    searchResults={searchResults}
+                    searchResultsFilter={searchResultsFilter}
+                    displayedCoverageId={displayedCoverageId}
+                    toggleSearch={controller.toggleSearch}
+                    toggleSearchResultsFilter={controller.toggleSearchResultsFilter}
+                    showMetadata={controller.showMetadata}
+                    toggleCoverageArea={controller.toggleCoverageArea}
+                    selectedLayers={selectedLayers}
+                    toggleLayerVisibility={controller.toggleLayerVisibility}/>
+            </>
+        }
+    </div>;
+};
+
+MetadataSearchContainer.propTypes = {
+    state: PropTypes.object,
+    controller: PropTypes.object,
+    advancedSearchOptions: PropTypes.object
+};
+
+export const renderMetadataSearchContainer = (state, controller, element) => {
+    const render = (state, controller) => {
+        ReactDOM.render(<MetadataSearchContainer state={state} controller={controller} />, element);
+    };
+
+    render(state, controller);
+    return {
+        update: render
+    };
+};
```

Comment:
This is unusual, but I guess it's needed because the container where it's rendered can be either on a jQuery based tab or jQuery based flyout?

Reply:
I thought so but I'm no quite sure if that's true... The whole bit seems a bit shady now, gotta dig a bit deeper into this.

Reply:
Turns out it really didn't have to be that complex. Refactored this bit and it still should work in botha cases - when the search bundle is available and when this is standalone.